utils/ansible.py

```diff
@@ -0,0 +1,281 @@
+            [provider.key].get('hostname', [])
+        }
+    elif script_type == 'users':
+        return {
+            'fullname': 'MIQUser',
+            'name': 'MIQU',
+            'password': 'smartvm',
+            'state': 'present',
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+    elif script_type == 'custom_attributes':
+        return {
+            'entity_type': 'provider',
+            'entity_name': conf.cfme_data.get('management_systems', {})
+            [provider.key].get('name', []),
+            'miq_url': config_formatter(),
+            'miq_username': conf.credentials['default'].username,
+            'miq_password': conf.credentials['default'].password,
+        }
+
+
+# TODO Avoid reading files every time
+def read_yml(script, value):
+    with open(yml_path + script + yml, 'r') as f:
+            doc = load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def get_yml_value(script, value):
+    with open(basic_yml_path + script + yml, 'r') as f:
+            doc = load(f)
+    return doc[0]['tasks'][0]['manageiq_provider'][value]
+
+
+def setup_basic_script(provider, script_type):
+    script_path = path.join(basic_yml_path, script_type + basic_script)
+    script_path_bk = path.join(basic_yml_path, script_type + basic_script_bk)
+    copyfile(script_path, script_path_bk)
+    with open(script_path, 'rw') as f:
+        doc = load(f)
+        values_dict = get_values_from_conf(provider, script_type)
+    for key in values_dict:
+        if script_type == 'providers':
+            doc[0]['tasks'][0]['manageiq_provider'][key] = values_dict[key]
+        elif script_type == 'users':
+            doc[0]['tasks'][0]['manageiq_user'][key] = values_dict[key]
+        elif script_type == 'custom_attributes':
+            doc[0]['tasks'][0]['manageiq_custom_attributes'][key] = values_dict[key]
+        with open(basic_yml_path + script_type + basic_script, 'w') as f:
+            f.write(dump(doc))
+
+
+def open_yml(script, script_type):
+    copyfile(basic_yml_path + "/" + script_type + basic_script,
+             basic_yml_path + script + yml)
+    with open(basic_yml_path + script + yml, 'rw') as f:
+        return load(f)
+
+
+def write_yml(script, doc):
+    with open(basic_yml_path + script + yml, 'w') as f:
+        f.write(dump(doc))
+
+
+def setup_ansible_script(provider, script, script_type=0, values_to_update=0, user_name=0):
+    # This function prepares the ansible scripts to work with the correct
+    # appliance configs that will be received from Jenkins
+    setup_basic_script(provider, script_type)
+    if script == 'add_provider':
+        copyfile(basic_yml_path + providers_basic_script, basic_yml_path + script + yml)
+
+    elif script == 'update_provider':
+        doc = open_yml(script, 'providers')
```

Comment:
**REQUIRED:** the repeated pattern of load, update, write needs to be extracted in order to avoid usage mistakes such as the too deep indentation of the write_yml function usage

calls could look like 

```python

update_script(script, 'providers', 'manageiq_provider', values_to_update)
update_script(script, 'providers', 'manageiq_provider', state='absent')
update_script(script, 'providers', 'manageiq_provider', state='absent', name=random_miq_user)

update_script(script, 'users', 'manageiq_user', values_to_update)